### PR TITLE
[SECURITY-2290] check permission as well

### PR DESCRIPTION
**Pull request by olamy**
Signed-off-by: Olivier Lamy <olamy@apache.org>

- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/jenkins/plugins/publish_over_ssh/descriptor/BapSshPublisherPluginDescriptor.java [hunk 1]
```diff
@@ -193,6 +193,7 @@ public BapSshHostConfigurationDescriptor getHostConfigurationDescriptor() {
 
     @RequirePOST
     public FormValidation doTestConnection(final StaplerRequest request, final StaplerResponse response) {
+        Jenkins.get().checkPermission(Jenkins.ADMINISTER););
```
Comment (Kevin-CB) [inline src/main/java/jenkins/plugins/publish_over_ssh/descriptor/BapSshPublisherPluginDescriptor.java:196]:
```suggestion
        Jenkins.get().checkPermission(Jenkins.ADMINISTER);
```

**Review by Kevin-CB — APPROVED** [review]
I tested this fix locally, the vulnerability is no longer present, however it is necessary to correct the 2 small copy paste typo.

**Comment by olamy** [comment]
oops my bad typo. good catch!

File: src/main/java/jenkins/plugins/publish_over_ssh/descriptor/BapSshHostConfigurationDescriptor.java [hunk 2]
```diff
@@ -81,12 +81,16 @@ public FormValidation doCheckTimeout(@QueryParameter final String value) {
         return FormValidation.validateNonNegativeInteger(value);
     }
 
+    @RequirePOST
```
Comment (daniel-beck) [inline src/main/java/jenkins/plugins/publish_over_ssh/descriptor/BapSshHostConfigurationDescriptor.java:84]: No side effects, so not strictly necessary, but doesn't hurt either assuming the UI controls send POST.

**Pull request merged**